File: tests/fixtures/go-project/service/user_service.go
package service

import (
	"fmt"
	"gofixture/repo"
)

type Notifier interface {
	Notify(message string) error
}

type Handler = func(string) error

const MaxUsers = 100

type UserService struct {
	repo *repo.UserRepo
}

func NewUserService(r *repo.UserRepo) *UserService {
	return &UserService{repo: r}
}

func (s *UserService) Save(name string) error {
	fmt.Println(name)
	return s.repo.Insert(name)
}

File: tests/fixtures/cpp-project/src/user_service.cpp
#include "user_repo.h"
#include <string>

namespace fixture {

class UserService {
public:
    int Save(const std::string& name);
private:
    UserRepo* repo_;
};

int UserService::Save(const std::string& name) {
    User* u = new User();
    return repo_->Insert(name);
}

void freeHelper(int a) {
}

}
